lib/api/dio_exceptions.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:dio/dio.dart';
+
+class DioExceptions implements Exception {
+  int code = 0;
+  String message = "";
+
+  DioExceptions({required this.code, required this.message});
+
+  factory DioExceptions.fromDioError(DioException dioException) {
+    switch (dioException.type) {
```

Comment:
Est-ce qu'il est prévu qu'il y ait plus que c'est 3 cas là et / ou que le retour change ?

Reply:
D'accord, car dans ce cas, pour simplifier le code, est-ceque tu pense que ça serait plus simple si on fait quelque chose comme ceci:

```suggestion
  List<DioExceptionType> dioExceptionTypes = [
    DioExceptionType.cancel,
    DioExceptionType.connectionTimeout,
    DioExceptionType.badResponse
  ];
  
  ...
  
  if (dioExceptionTypes.contains(dioException.type)) {
      return DioExceptions(
          code: dioException.response!.statusCode!,
          message: dioException.response?.data['message']
      );
    }

    return DioExceptions(code: 500, message: dioException.response?.data['message']);
```

Reply:
yep ça fonctionne ! je le change